app/scripts/directives/resources.js

```diff
@@ -1,6 +1,75 @@
 'use strict';
 
 angular.module('openshiftConsole')
+  .directive('containerStatuses', function($filter) {
+    return {
+      restrict: 'E',
+      scope: {
+        containerStatuses: '=',
+        startDebugTerminal: '=?',
+        detailed: '=?'
+      },
+      templateUrl: 'views/_container-statuses.html',
+      link: function(scope) {
+        scope.hasDebugTerminal = angular.isFunction(scope.startDebugTerminal);
+
+        scope.podContainerStatuses = _.get(scope.containerStatuses, 'status.containerStatuses', []);
+        scope.podContainerInitStatuses = _.get(scope.containerStatuses, 'status.initContainerStatuses', []);
+
+        scope.hasContainerFailed = function (containerStatus) {
+          return $filter('isContainerFailed')(containerStatus) !== false;
+        };
+
+        scope.haveContainersFailed = function(containerStatuses) {
```

Comment:
Same comment here. Doesn't need to be a scope function. `_.some` makes this easier.

```js
var isContainerFailed = $filter('isContainerFailed');
var haveContainersFailed = function(containerStatuses) {
  return _.some(containerStatuses, isContainerFailed);
};
```

https://lodash.com/docs/3.10.1#some